File: lab4/lab4_Java/src/Repository.java
import java.util.List;

public interface Repository {
    List<Tour> getAll();
    Tour getById(int id);
    boolean addTour(Tour tour);
    boolean deleteTour(int id);
    List<Tour> getByTourName(String tourName);
    List<Tour> getByTourNameAndPaid(String tourName, boolean paid);
}

File: lab4/lab4_Kotlin/src/Repository.kt
interface Repository {
    fun getAll(): List<Tour>
    fun getById(id: Int): Tour
    fun addTour(tour: Tour): Boolean
    fun deleteTour(id: Int): Boolean
    fun getByTourName(tourName: String): List<Tour>
    fun getByTourNameAndPaid(tourName: String, paid: Boolean): List<Tour>
}